api/src/couchdb/attachmentExport.ts

```diff
@@ -0,0 +1,503 @@
+          }
+
+          // Advance to next record
+          const next = await iterator.next();
+          record = next.record;
+          done = next.done;
+        }
+
+        /**
+         * Wait for at least one stream to complete before continuing.
+         * This prevents the loop from spinning when at max concurrency.
+         */
+        if (activeStreams.size > 0) {
+          await Promise.race(activeStreams);
+        }
+      }
+    }
+
+    // Handle edge case: no attachments found in any records
+    if (fileCount === 0) {
+      console.log('[ZIP] No attachments found, aborting archive creation');
+      archive.abort();
+      return;
+    }
+
+    /**
+     * CRITICAL: Finalize the archive only after ALL streams have completed.
+     * Finalizing too early will result in a corrupted ZIP file.
+     */
+    console.log(
+      `[ZIP] All files processed. Finalizing archive with ${fileCount} files`
+    );
+    await archive.finalize();
+
+    logMemory(
+      'COMPLETE',
+      `Successfully archived ${fileCount} files from ${recordCount} records`
+    );
+  } catch (error) {
+    console.error('[ZIP] Fatal error during archive creation:', error);
+    throw new Error(
+      `Failed to create ZIP archive: ${error instanceof Error ? error.message : 'Unknown error'}`
+    );
+  }
+};
+
+/**
+ * Creates a notebook record iterator configured for efficient attachment streaming.
+ *
+ * IMPORTANT: Sets includeAttachments=false to avoid loading attachment binary data
+ * into memory during iteration. We only need metadata (attachment_id, filename, type)
+ * which is included in the record structure regardless of this flag.
+ *
+ * @param projectId - The project identifier
+ * @param viewID - The view identifier for filtering
+ * @param includeAttachments - Whether to load full attachment data (always false for streaming)
+ * @returns An async iterator over hydrated records
+ */
+async function createNotebookRecordIterator(
+  projectId: ProjectID,
+  viewID: string,
+  includeAttachments = false
+) {
+  const dataDb = await getDataDb(projectId);
+  const uiSpecification = await getProjectUIModel(projectId);
+
+  return notebookRecordIterator({
+    projectId,
+    filterDeleted: true, // Exclude deleted records from export
+    dataDb,
+    uiSpecification,
+    viewID,
+    includeAttachments, // False to prevent memory bloat from large attachments
+  });
+}
+
+/**
+ * Creates and configures an archiver instance with proper error handling.
+ *
+ * Configuration:
+ * - Format: ZIP
+ * - Compression level: 9 (maximum compression)
+ * - Error handling: Warnings logged, errors thrown
+ *
+ * @param outputStream - The destination stream for the ZIP archive
+ * @returns Configured archiver instance ready to accept files
+ */
+function createConfiguredArchive(
+  outputStream: NodeJS.WritableStream
+): archiver.Archiver {
+  const archive = archiver('zip', {
+    zlib: {level: 9}, // Maximum compression for smaller file sizes
```

Comment:
Just did some tests, on a 6.5G collection of images, level 9 compression only shrinks it to 6.4G, level 1 does the same.  Since JPEG images are already compressed we might be better using a lower level here to make it faster/less resource hungry.